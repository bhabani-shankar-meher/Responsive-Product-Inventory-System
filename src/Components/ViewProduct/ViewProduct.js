import { useEffect, useState } from "react";
import { useParams, useNavigate } from "react-router-dom";
import { Button } from "@mui/material";
import Box from "@mui/material/Box";
import Grid from "@mui/material/Grid";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemText from "@mui/material/ListItemText";
import Typography from "@mui/material/Typography";
const ViewProduct = () => {
  const { id } = useParams();
  let navigate = useNavigate();
  const [selectedItem, setSelectedItem] = useState("");
  useEffect(() => {
    fetch(`http://localhost:4000/products/${id}`).then((response) =>
      response.json().then((data) => setSelectedItem(data))
    );
  }, [id]);

  useEffect(() => {
    selectedItem &&
      (selectedItem.noOfClicked = selectedItem.noOfClicked + 1);
    selectedItem &&
      fetch(`http://localhost:4000/products/${id}`, {
        method: "PATCH",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(selectedItem),
      });
  }, [selectedItem]);

  return (
    <Box sx={{ width: "50%", marginLeft: "25%" }}>
      <Grid item xs={12} md={12} style={{ boxShadow: "5px 5px 15px -1px rgb(0 0 0 / 22%)", background: "white", marginTop: "2rem" }}>
        <List sx={{ width: "100%" }}>
          {selectedItem && (
            <ListItem alignItems="center">
              <ListItemText
                style={{ textAlign: "center" }}
                primary={`Product Name : ${selectedItem.productName}`}
                secondary={
                  <>
                    <Typography
                      sx={{ display: "block" }}
                      mt={2}
                      component="h6"
                      variant="h6"
                      color="text.primary"
                    >
                      {`Quantity : ${selectedItem.quantity}`}
                    </Typography>
                    <Typography
                      sx={{ display: "block" }}
                      mt={2}
                      variant="p"
                      color="text.primary"
                    >
                      {`Price : ${selectedItem.price}`}
                    </Typography>
                    <Typography
                      sx={{ display: "block" }}
                      mt={2}
                      variant="p"
                      color="text.primary"
                    >
                      {`Created Date : ${selectedItem.createdDate}`}
                    </Typography>
                    <Typography
                      sx={{ display: "block" }}
                      mt={2}
                      variant="p"
                      color="text.primary"
                    >
                      {`Updated Date : ${selectedItem.updatedDate}`}
                    </Typography>
                    <Typography
                      sx={{ display: "block" }}
                      mt={2}
                      variant="p"
                      color="text.primary"
                    >
                      {`Viewed : ${selectedItem.noOfClicked}`}
                    </Typography>
                  </>
                }
              />
            </ListItem>
          )}
        </List>
        <Grid item xs={12} md={12} mt={5} style={{ display: "flex", justifyContent: "center", paddingBottom: "2rem" }}>
          <Button
            variant="contained"
            color="primary"
            style={{ width: "10rem" }}
            onClick={() => { navigate(`/update-product/${selectedItem.id}`, { replace: true }) }}
          >
            Update
          </Button>
          <Button
            variant="outlined"
            color="secondary"
            style={{ width: "10rem", marginLeft: "2rem" }}
            onClick={() => navigate("/", { replace: true })}
          >
            Back
          </Button>
        </Grid>
      </Grid>
    </Box>
  );
};

export default ViewProduct;
