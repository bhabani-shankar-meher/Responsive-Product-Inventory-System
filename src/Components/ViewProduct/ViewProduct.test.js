import ViewProduct from "./ViewProduct";
import { render, screen } from "@testing-library/react";
import { create } from "react-test-renderer";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => jest.fn(),
}));

describe("VIEWPRODUCT PAGE SNAPSHOT", () => {
    test("ViewProduct Page Snapshot Executed Successfully", () => {
        const viewProductSnapshot = create(<ViewProduct />);
        expect(viewProductSnapshot).toMatchSnapshot();
    });
});
describe("VIEWPRODUCT PAGE TEST", () => {
    test("Update Button Test", () => {
        render(<ViewProduct />);
        expect(
            screen.getByRole("button", {
                name: /update/i,
            })
        ).toBeInTheDocument();
    });
    test("Back Button Test", () => {
        render(<ViewProduct />);
        expect(
            screen.getByRole("button", {
                name: /back/i,
            })
        ).toBeInTheDocument();
    });
});