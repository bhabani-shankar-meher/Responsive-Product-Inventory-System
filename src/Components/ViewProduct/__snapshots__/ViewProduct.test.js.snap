// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VIEWPRODUCT PAGE SNAPSHOT ViewProduct Page Snapshot Executed Successfully 1`] = `
<div
  className="MuiBox-root css-15gx3v1"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-13i4rnv-MuiGrid-root"
    style={
      Object {
        "background": "white",
        "boxShadow": "5px 5px 15px -1px rgb(0 0 0 / 22%)",
        "marginTop": "2rem",
      }
    }
  >
    <ul
      className="MuiList-root MuiList-padding css-1lbrbe2-MuiList-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-vkfc0g-MuiGrid-root"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "paddingBottom": "2rem",
        }
      }
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "10rem",
          }
        }
        tabIndex={0}
        type="button"
      >
        Update
      </button>
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginLeft": "2rem",
            "width": "10rem",
          }
        }
        tabIndex={0}
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</div>
`;
